packages/lib/services/interop/InteropService_Exporter_Md.ts

```diff
@@ -146,6 +146,9 @@ export default class InteropService_Exporter_Md extends InteropService_Exporter_
 			fileName = friendlySafeFilename(resource.title, null, true);
 		}
 
+		// Markdown link destination should not contain space characters
+		fileName = fileName.replace(/\s/g, '_');
```

Comment:
I think this fix is not right. Urls with spaces should be supported, so I think it's the code that add the url to the Markdown content that should be escaped. We have utility functions in markdownUtils for this

Reply:
I made some changes, please tell me if it is what you had in mind.

Reply:
It's not right either, it should be escaped when it's inserted in the doc

Reply:
So you're suggesting the filename should be escaped when adding an image (or another resource) to a note?
Actually I don't think we need it because from what I've seen we are only using resource id in the note. The issue only happens when we export the note and we use the original filename.

Reply:
As I said I'm not sure. I don't know why it's not currently escaped, there could be a good reason, but I'd need to look at the code

Reply:
So the fix is correct, however as mentioned the data should only be escaped when the text is added to the document, which would be here: https://github.com/laurent22/joplin/blob/9aa59360c1ef41d35b75b0eb7d53b6a3a6c0ab0f/packages/lib/services/interop/InteropService_Exporter_Md.ts#L75

We only escape when needed, not before. This is to avoid issues with data being double-escaped or code using escaped data, when it expects to deal with unescape one, which can lead to bugs and security issues.

Are you ok with making this change?